bot/cogs/help_channels.py

```diff
@@ -0,0 +1,672 @@
+        """Make the `channel` dormant."""
+        log.info(f"Moving #{channel} ({channel.id}) to the Dormant category.")
+
+        await channel.edit(
+            category=self.dormant_category,
+            sync_permissions=True,
+            topic=DORMANT_TOPIC,
+            position=self.get_position(channel, self.dormant_category),
+        )
+
+        log.trace(f"Position of #{channel} ({channel.id}) is actually {channel.position}.")
+
+        log.trace(f"Sending dormant message for #{channel} ({channel.id}).")
+        embed = discord.Embed(description=DORMANT_MSG)
+        await channel.send(embed=embed)
+
+        log.trace(f"Pushing #{channel} ({channel.id}) into the channel queue.")
+        self.channel_queue.put_nowait(channel)
+
+    async def move_to_in_use(self, channel: discord.TextChannel) -> None:
+        """Make a channel in-use and schedule it to be made dormant."""
+        log.info(f"Moving #{channel} ({channel.id}) to the In Use category.")
+
+        await channel.edit(
+            category=self.in_use_category,
+            sync_permissions=True,
+            topic=IN_USE_TOPIC,
+            position=0,
+        )
+
+        timeout = constants.HelpChannels.idle_minutes * 60
+
+        log.trace(f"Scheduling #{channel} ({channel.id}) to become dormant in {timeout} sec.")
+        data = TaskData(timeout, self.move_idle_channel(channel))
+        self.schedule_task(channel.id, data)
+
+    async def notify(self) -> None:
+        """
+        Send a message notifying about a lack of available help channels.
+
+        Configuration:
+
+        * `HelpChannels.notify` - toggle notifications
+        * `HelpChannels.notify_channel` - destination channel for notifications
+        * `HelpChannels.notify_minutes` - minimum interval between notifications
+        * `HelpChannels.notify_roles` - roles mentioned in notifications
+        """
+        if not constants.HelpChannels.notify:
+            return
+
+        log.trace("Notifying about lack of channels.")
+
+        if self.last_notification:
+            elapsed = (datetime.utcnow() - self.last_notification).seconds
+            minimum_interval = constants.HelpChannels.notify_minutes * 60
+            should_send = elapsed >= minimum_interval
+        else:
+            should_send = True
+
+        if not should_send:
+            log.trace("Notification not sent because it's too recent since the previous one.")
+            return
+
+        try:
+            log.trace("Sending notification message.")
+
+            channel = self.bot.get_channel(constants.HelpChannels.notify_channel)
+            mentions = " ".join(f"<@&{role}>" for role in constants.HelpChannels.notify_roles)
+
+            message = await channel.send(
+                f"{mentions} A new available help channel is needed but there "
+                f"are no more dormant ones. Consider freeing up some in-use channels manually by "
```

Comment:
this doesn't really have to be an f-string, but on the other hand, the f makes it line up neatly with the others so I'm not actually sure which I prefer. Maybe we just leave it.

Reply:
I prefer to leave it.